drivers/pwm/pwm_nxp_s32_emios.c

```diff
@@ -9,7 +9,10 @@
 #include <zephyr/drivers/pinctrl.h>
 #include <zephyr/drivers/clock_control.h>
 #include <Emios_Mcl_Ip.h>
+
+#ifdef CONFIG_PWM_NXP_S32_EMIOS_PULSE_GEN
```

Comment:
do we really need to build everything conditionally? cannot we assume that PWM generation is always built, as in the rest of the PWM drivers?

Reply:
Still need conditional because of the glue code. For example no channel is configured for PWM generation such as this https://github.com/zephyrproject-rtos/hal_nxp/pull/277/files#diff-34cb08b7d966e3016d192b073a8796259280fa3cfca2861ae13eb754bf145216R142 will expose to 0 then causes some arrays declared with no element --> warning when build. I've updated to don't need user need to set any Kconfig or the like